### PR TITLE
simon/issue#137/capitalI

### PR DESCRIPTION
**Pull request by psi77**
Hopefully fixed the capital I issue by appending '<space><period>' to the end of every sentence if there isn't currently any punctuation at the end of the sentence (such as a ? or !)
This shouldn't interfere with inflection for existing questions etc.

Also applied the same logic to the speak now stuff.

### CODE REVIEW

File: src/app/speechbar.service.spec.ts [hunk 1]
```diff
@@ -318,3 +318,22 @@ describe('SpeechbarService.ButtonFacade', () => {
...
+  it('should append period if no punctuation', inject([SpeechbarService], (service: SpeechbarService) => {
+    const vocals = ['I', 'hello'];
+    expect(service.buildSentence(vocals)).toBe('I hello .');
+  }));
+
+  it('should not append period if punctuation', inject([SpeechbarService], (service: SpeechbarService) => {
+    let vocals = ['I', 'hello?'];
+    expect(service.buildSentence(vocals)).toBe('I hello?');
+    vocals = ['I', 'hello!'];
+    expect(service.buildSentence(vocals)).toBe('I hello!');
```
Comment (blaise17) [inline src/app/speechbar.service.spec.ts]: Might be nice to have a test for a period on the end as well?

File: src/app/speechbar.service.ts [hunk 2]
```diff
@@ -126,19 +126,29 @@ export class SpeechbarService {
...
+    const endsWithPunc = lastChar === '?' || lastChar === '!' || lastChar === '.';
+    return endsWithPunc ? ret : ret + ' .';
+  }
+
   sayButton(button: Button) {
     // we do want to queue these up, so no speaking check
     // we also want the speak button to stay enabled for queing up the actual message, so no listener updates
     const msg = this.createUtterance();
-    msg.text = button.getVocalization();
+    msg.text = this.buildSentence([button.getVocalization()]);
```
Comment (blaise17) [inline src/app/speechbar.service.ts]: Nice reuse :D

File: src/app/speechbar.service.spec.ts (hunk 1 above)
Comment (psi77) [reply src/app/speechbar.service.spec.ts]: Yes. It would. I don't know why I didn't add it.
I miss parametrized tests.
I'll add it

Comment (psi77) [reply src/app/speechbar.service.spec.ts]: oh man.  Now you need to re-review!

**Pull request merged**